sunbeam-python/sunbeam/plugins/validation/plugin.py

```diff
@@ -0,0 +1,554 @@
+                    )
+                )
+            except (ActionFailedException, UnitNotFoundException) as e:
+                raise click.ClickException(str(e))
+
+            if action_result.get("return-code", 0) > 1:
+                message = f"Unable to run action: {action_name}"
+                raise click.ClickException(message)
+
+            return action_result
+
+    def _check_file_exist_in_tempest_container(self, filename: str) -> bool:
+        """Check if file exist in tempest container."""
+        unit = self._get_tempest_leader_unit()
+        # Note: this is a workaround to run command to payload container
+        # since python-libjuju does not support such feature. See related
+        # bug: https://github.com/juju/python-libjuju/issues/1029
+        try:
+            subprocess.run(
+                [
+                    "juju",
+                    "ssh",
+                    "--container",
+                    TEMPEST_CONTAINER_NAME,
+                    unit,
+                    "ls",
+                    TEMPEST_VALIDATION_RESULT,
+                ],
+                check=True,
+                timeout=30,  # 30 seconds should be enough for `ls`
+                capture_output=True,
+            )
+        except subprocess.CalledProcessError:
+            return False
+        except subprocess.TimeoutExpired:
+            raise click.ClickException(f"Timed out checking {filename}")
+        return True
+
+    def _copy_file_from_tempest_container(self, source: str, destination: str) -> None:
+        """Copy file from tempest container."""
+        unit = self._get_tempest_leader_unit()
+        progress_message = (
+            f"Copying {source} from "
+            f"{TEMPEST_APP_NAME} ({TEMPEST_CONTAINER_NAME}) "
+            f"to {destination} ..."
+        )
+        with console.status(progress_message):
+            # Note: this is a workaround to run command to payload container
+            # since python-libjuju does not support such feature. See related
+            # bug: https://github.com/juju/python-libjuju/issues/1029
+            if Path(destination).is_dir():
+                # juju scp does not allow directory as destination
+                destination = str(Path(destination, Path(source).name))
+            try:
+                subprocess.run(
+                    [
+                        "juju",
+                        "scp",
+                        "--container",
+                        TEMPEST_CONTAINER_NAME,
+                        f"{unit}:{source}",
+                        destination,
+                    ],
+                    check=True,
+                    timeout=60,  # 60 seconds should be enough for copying a file
+                )
+            except (subprocess.CalledProcessError, subprocess.TimeoutExpired) as e:
+                raise click.ClickException(str(e))
+
+    def _configure_preflight_check(self) -> False:
```

Comment:
Return should be bool

Reply:
Thanks for spotting it. no idea why it was `False`.